src/test/java/integration/IntegrationTest.java

```diff
@@ -3,563 +3,1428 @@
+            Assert.assertNotEquals("Authenticated method succeeded with no auth info", 200, response.getStatus());
+
+            waiter.release();
+        }, errHandler);
+
+        ws3.sendRequest(req);
+        if (!waiter.tryAcquire(5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
+    }
+
+    private void testUserLookup() throws ConnectException, InterruptedException {
+        Semaphore waiter = new Semaphore(0);
+
+        req = new UserLookupRequest(new String[]{user2ID}).getRequest(response -> {
+            Assert.assertEquals("Failed to lookup user 2", 200, response.getStatus());
+
+            Assert.assertEquals("Incorrect number of users returned when looking up user 2", 1, ((UserLookupResponse) response.getData()).getUsers().length);
+            Assert.assertEquals("Incorrect first name returned when looking up user 2", user2FirstName, ((UserLookupResponse) response.getData()).getUsers()[0].getFirstName());
+            Assert.assertEquals("Incorrect last name returned when looking up user 2", user2LastName, ((UserLookupResponse) response.getData()).getUsers()[0].getLastName());
+            Assert.assertEquals("Incorrect email returned when looking up user 2", user2Email, ((UserLookupResponse) response.getData()).getUsers()[0].getEmail());
+            Assert.assertEquals("Incorrect username returned when looking up user 2", user2ID, ((UserLookupResponse) response.getData()).getUsers()[0].getUsername());
+
+            waiter.release();
+        }, errHandler);
+
+        ws1.sendRequest(req);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
+
+        req = new UserLookupRequest(new String[]{user1ID}).getRequest(response -> {
+            Assert.assertEquals("Failed to lookup user 1", 200, response.getStatus());
+
+            Assert.assertEquals("Incorrect number of users returned when looking up user 1", 1, ((UserLookupResponse) response.getData()).getUsers().length);
+            Assert.assertEquals("Incorrect first name returned when looking up user 1", user1FirstName, ((UserLookupResponse) response.getData()).getUsers()[0].getFirstName());
+            Assert.assertEquals("Incorrect last name returned when looking up user 1", user1LastName, ((UserLookupResponse) response.getData()).getUsers()[0].getLastName());
+            Assert.assertEquals("Incorrect email returned when looking up user 1", user1Email, ((UserLookupResponse) response.getData()).getUsers()[0].getEmail());
+            Assert.assertEquals("Incorrect username returned when looking up user 1", user1ID, ((UserLookupResponse) response.getData()).getUsers()[0].getUsername());
+
+            waiter.release();
+        }, errHandler);
+
+        ws2.sendRequest(req);
+        if (!waiter.tryAcquire(1, 5, TimeUnit.SECONDS)) {
+            Assert.fail("Acquire timed out");
+        }
+    }
```

Comment:
That's a good point, I forgot about the potential for partial failure. Might want to also do the same kind of check for the other methods that return a list of something (User.Projects, Project.Lookup)